addOns/pscanrulesAlpha/src/test/java/org/zaproxy/zap/extension/pscanrulesAlpha/BigRedirectsScannerTest.java

```diff
@@ -0,0 +1,123 @@
+import org.parosproxy.paros.network.HttpRequestHeader;
+
+public class BigRedirectsScannerTest extends PassiveScannerTest<BigRedirectsScanner> {
+    private static final String URI = "http://example.com";
+
+    protected HttpMessage msg;
+
+    @Override
+    protected BigRedirectsScanner createScanner() {
+        return new BigRedirectsScanner();
+    }
+
+    @Test
+    public void GivenRedirectWithSmallBodyThenItRaisesNoAlert() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize()]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void GivenRedirectHeadersWithLargeBodyThenAlertRaised() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(1));
+        assertBigRedirectAlertAttributes(alertsRaised.get(0));
+    }
+
+    @Test
+    public void GivenNotModifiedStatusCodeWithLargeBodyThenNoAlertRaised() {
+        msg.getResponseHeader().setStatusCode(304);
+        msg.getResponseHeader().setHeader("Location", URI);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Test
+    public void GivenRedirectStatusCodeWithoutLocationHeaderThenNoAlertRaised() {
+        msg.getResponseHeader().setStatusCode(300);
+        msg.setResponseBody(new byte[allowableBodySize() + 1]);
+
+        rule.scanHttpResponseReceive(msg, -1, new Source(""));
+
+        assertThat(alertsRaised.size(), is(0));
+    }
+
+    @Before
+    public void CreateHttpMessage() throws IOException {
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI(URI, false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+    }
+
+    private void assertBigRedirectAlertAttributes(Alert alert) {
+        assertThat(alert.getRisk(), is(Alert.RISK_LOW));
+        assertThat(alert.getConfidence(), is(Alert.CONFIDENCE_MEDIUM));
+        assertThat(alert.getName(), is(localisedString("name")));
+        assertThat(alert.getDescription(), is(localisedString("desc")));
+        assertThat(alert.getSolution(), is(localisedString("soln")));
+        assertThat(alert.getReference(), is(localisedString("refs")));
+        assertThat(alert.getOtherInfo(), is(expectedExtraInfo()));
+        assertThat(alert.getCweId(), is(201));
+        assertThat(alert.getWascId(), is(13));
+        assertThat(alert.getUri(), is(URI));
+    }
+
+    private String expectedExtraInfo() {
+        int bodySize = allowableBodySize();
```

Comment:
Is there a reason to store this locally and not just use `allowableBodySize()` in the statement below?

Reply:
With the longer name it causes the linter to break the below line into 2.

Thought these 2 lines looked nicer, but it's more of a stylistic thing.

Reply:
Fair enough, not a major issue.